lib/Extractor/Solver.cpp

```diff
@@ -114,6 +132,20 @@ class BaseSolver : public Solver {
       std::vector<llvm::APInt> ModelVals;
       Inst *I = IC.createVar(LHS->Width, "constant");
       InstMapping Mapping(LHS, I);
+
+      if (EnableAliveTest) {
+        Inst *Ante = IC.getConst(APInt(1, true));
+        for (auto PC : PCs ) {
+          Inst *Eq = IC.getInst(Inst::Eq, 1, {PC.LHS, PC.RHS});
+          Ante = IC.getInst(Inst::And, 1, {Ante, Eq});
+        }
+        bool IsValid = isTransformationValid(Mapping.LHS, Mapping.RHS, Ante);
+        if (IsValid) {
+          RHS = I;
+        }
+        return std::error_code(); // TODO: Propagate errors from Alive backend
```

Comment:
same here, don't just return. if synthesizing an integer fails, we still want to fall down to exhaustive syntheiss (or CEGIS or whatever)